sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -233,7 +233,7 @@ class FileStreamSource(
     }
 
     val files = allFiles.sortBy(_.getModificationTime)(fileSortOrder).map { status =>
-      (status.getPath.toUri.toString, status.getModificationTime)
+      (status.getPath.toUri.getPath, status.getModificationTime)
```

Comment:
why not `status.getPath.toString`?

Reply:
ping @xysun ^

Reply:
The correct fix is fixing this line: https://github.com/xysun/spark/blob/4f5979a72ce9cb36a3327e79b8592b9e42bdf5af/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala#L168

It should be `files.map(new Path(new URI(_.path)).toString)`.

Reply:
FYI, `getPath` will drop scheme and credentials.

Reply:
Wait .. @zsxwing do you mean `getPath` (`org.apache.hadoop.fs.Path`) from `org.apache.hadoop.fs.FileStatus` drops the scheme and credentials?

Here seems `Seq[org.apache.hadoop.fs.FileStatus]` and the code you pointed out looks `Array[org.apache.spark.sql.execution.streaming.FileStreamSource.FileEntry]`.

Reply:
Ohaa, `getPath` from `java.net.URI` in the proposed fix drops them. Sure.

Reply:
Thanks. I'll update the commit.